chromium/rules.js

```diff
@@ -512,7 +511,7 @@ RuleSets.prototype = {
     // observed and the domain blacklisted, a cookie might already have been
     // flagged as secure.
 
-    if (domainBlacklist.has(domain)) {
+    if (background.domainBlacklist.has(domain)) {
```

Comment:
IIUC, this still relies on global `background` variable which makes it impossible to import just `rules` in Node without defining that one first. I tried to fix that with https://github.com/EFForg/https-everywhere/pull/12556 by moving rules options to rules itself, making it self-contained, but there are conflicts now.

Reply:
Or I can make a PR for rules.js adding a dummy `background` object if there is no real one.

Reply:
@koops76 Exactly. As I said, not directly importable in a modular fashion without polluting globals (just like before, and was fixed in a PR I linked).

Reply:
@RReverser can you verify that this is indeed an issue? I am able to import this.

Before this PR, `domainBlacklist` was already `undefined` when this is imported by node. So this should not change anything.

Reply:
I am able to import `rules.js` fine, and the `rewriter.js` script runs fine for me with cursory testing.

Reply:
@cowlicks Yeah but can you actually use it? I mean, as soon as it goes into one of code paths where it tries to access `background` variable, it will fail with ReferenceError (yay JS, but anyway ESLint would've warned about it if it wouldn't be in the globals array in config).